modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ModuleOptionsComponent.java

```diff
@@ -572,34 +572,39 @@ public void updateResourceConfiguration(ConfigurationUpdateReport report) {
-                //set this new state passed in as state of the attribute.
-                LinkedHashMap<String, Object> currentOptions = new LinkedHashMap<String, Object>();
-                for (String propertyKey : mapType.getMap().keySet()) {
-                    currentOptions.put(propertyKey, ((PropertySimple) mapType.get(propertyKey)).getStringValue());
-                }
+                try{
+                    int index = path.indexOf(loginModuleIdentifier);
+                
+                    String loginModuleIndex = path.substring(index + loginModuleIdentifier.length());
+                    String[] split = loginModuleIndex.split(",");
+                    int lmi = Integer.valueOf(split[0]);//Ex 0,1,30
+                    Value valueObject = currentAttributeState.get(lmi);
+
+                    //retrieve current config changes    
+                    Configuration conf = report.getConfiguration();
+                    //list current conf stated being defined.
+                    String mapKey = (String) conf.getMap().keySet().toArray()[0];
+                    PropertyMap mapType = (PropertyMap) conf.getMap().get(mapKey);
+                    //insert update logic.
+                    //set this new state passed in as state of the attribute.
+                    LinkedHashMap<String, Object> currentOptions = new LinkedHashMap<String, Object>();
+                    for (String propertyKey : mapType.getMap().keySet()) {
+                        currentOptions.put(propertyKey, ((PropertySimple) mapType.get(propertyKey)).getStringValue());
+                    }
 
-                //make these settings the new state for this part of the attribute.
-                valueObject.setOptions(currentOptions);
-                currentAttributeState.set(lmi, valueObject);
+                    //make these settings the new state for this part of the attribute.
+                    valueObject.setOptions(currentOptions);
+                    currentAttributeState.set(lmi, valueObject);
 
-                Operation write = new WriteAttribute(address);
-                write.addAdditionalProperty("name", attribute);//attribute to execute on
+                    Operation write = new WriteAttribute(address);
+                    write.addAdditionalProperty("name", attribute);//attribute to execute on
 
-                //now complete the write operation by updating the value
-                write.addAdditionalProperty("value", (Object) currentAttributeState);
-                executeWriteAndGenerateAs7ServerUpdate(report, conf, write);
+                    //now complete the write operation by updating the value
+                    write.addAdditionalProperty("value", (Object) currentAttributeState);
+                    executeWriteAndGenerateAs7ServerUpdate(report, conf, write);
+                } catch(NumberFormatException e) {
+                    // Ignore it, the value just wasn't available
```

Comment:
No, it should not (it already does, lets avoid that spam), as we do not have a process for these modules that are not supporting our indexing scheme. So ignoring them is fine.

However, the NumberFormatException catches too long section in my opinion, hiding potentially some other issues than just the index parsing. So the area of try should be reduced.